src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const objects = [];
+  let newObj = { ...state };
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        newObj = { ...newObj };
```

Comment:
there is no need to copy object to itself

Reply:
it changed the object without coping
last 2 tests failed without this line

Reply:
in this case, you can delete all 3 `objects.push(newObj);` and create `objects.push({ ...newObj });` before the last `return` statement of this function